istioctl/pkg/waypoint/waypoint.go

```diff
@@ -480,13 +481,24 @@ func deleteWaypoints(cmd *cobra.Command, kubeClient kube.CLIClient, namespace st
 			if gw.Spec.GatewayClassName != constants.WaypointGatewayClassName {
 				continue
 			}
-			names = append(names, gw.Name)
+			nameList = append(nameList, gw.Name)
+		}
+	} else {
+		for _, name := range names {
+			gw, err := kubeClient.GatewayAPI().GatewayV1().Gateways(namespace).Get(context.Background(), name, metav1.GetOptions{})
```

Comment:
I prefer list with filtering instead of multiple get if we have several names specified here.

Reply:
I have tried this idea, but found that it does not work when using `FieldSelector` of `ListOptions` to filter `Spec.GatewayClassName` of Gateway, like that:

```go
		waypoints, err := kubeClient.GatewayAPI().GatewayV1().Gateways(namespace).
			List(context.Background(), metav1.ListOptions{
				LabelSelector: selector,
				FieldSelector: "spec.gatewayClassName=istio-waypoint",
			})
```

 Do you have any suggestions? Thanks

Reply:
Or can we list all gateways in this namespace, and then filter by name/GatewayClassName? @hanxiaop

Reply:
> can we list all gateways in this namespace, and then filter by name/GatewayClassName

Sorry for missing the issue. This works for me.

Reply:
updated, please review again, thanks